eslint.js

```diff
@@ -383,49 +385,83 @@ module.exports = {
         'yield-star-spacing': 'off',
 
         // Import
+
+        // Static analysis
         'import/no-unresolved': 'error',
         'import/named': 'error',
         'import/default': 'error',
         'import/namespace': 'error',
+        'import/no-restricted-paths': 'off',
+        'import/no-absolute-path': 'error',
+        'import/no-dynamic-require': 'off',
+        'import/no-internal-modules': 'off',
+        'import/no-webpack-loader-syntax': 'off',
+
+        // Helpful warnings
         'import/export': 'error',
         'import/no-named-as-default': 'error',
         'import/no-named-as-default-member': 'error',
         'import/no-deprecated': 'warn',
-        'import/no-extraneous-dependencies': 'error',
+        'import/no-extraneous-dependencies': 'off',
         'import/no-mutable-exports': 'error',
+
+        // Module systems
+        'import/unambiguous': 'off',
+        'import/no-commonjs': 'off',
         'import/no-amd': 'error',
-        'import/imports-first': 'error',
+        'import/no-nodejs-modules': 'off',
+
+        // Style guide
+        'import/first': 'error',
         'import/no-duplicates': 'error',
+        'import/no-namespace': 'off',
+        'import/extensions': ['error', {js: 'never'}],
+        'import/order': 'off',
+        'import/newline-after-import': 'error',
         'import/prefer-default-export': 'error',
+        'import/max-dependencies': 'off',
+        'import/no-unassigned-import': 'off',
+        'import/no-named-default': 'error',
+        'import/no-anonymous-default-export': 'off',
 
         // Promise
         'promise/catch-or-return': 'error',
-        'promise/always-return': 'error',
+        'promise/always-return': 'off',
         'promise/param-names': 'error',
         'promise/no-native': 'off',
+        'promise/no-return-wrap': 'off',
+        'promise/no-nesting': 'error',
+        'promise/no-promise-in-callback': 'error',
+        'promise/no-callback-in-promise': 'error',
+        'promise/avoid-new': 'off',
+        'promise/prefer-await-to-then': 'error',
+        'promise/prefer-await-to-callbacks': 'error',
 
         // Flow type
         'flowtype/boolean-style': ['error', 'boolean'],
         'flowtype/define-flow-type': 'error',
         'flowtype/delimiter-dangle': ['error', 'only-multiline'],
         'flowtype/generic-spacing': ['error', 'never'],
         'flowtype/no-dupe-keys': 'error',
+        'flowtype/no-primitive-constructor-types': 'error',
         'flowtype/no-weak-types': 'off',
-        'flowtype/require-parameter-type': 'off',
+        'flowtype/object-type-delimiter': 'error',
+        'flowtype/require-parameter-type': 'error',
         'flowtype/require-return-type': 'off',
```

Comment:
I'm not sure whether to turn this one off or not because it might be a touch _too_ aggressive; it does what it says on the tin but it's demanding return types for all manner of functions, describe/it blocks included.